### DISCUSSION TITLE
Windows App Using Edge WebView2

### DISCUSSION THREAD

**Discussion by RehanSaeed** [GoogleChrome/lighthouse#16286] [Q&A]
I'm writing a Windows desktop app using a [WebView2](https://learn.microsoft.com/en-gb/microsoft-edge/webview2/?form=MA13LH) component which uses Edge under the hood. I can't run the app in a browser because it needs certain interop API's from the host Windows application to function. The desktop app also helps me handle auth and everything.

How can I point lighthouse (probably the CLI tool) at the WebView2 inside my Windows desktop app? I've tried using a remote debugging port and trying to tell lighthouse to reuse the page but it keeps opening up a new browser window, failing auth etc and completing its report.

```powershell
lighthouse https://localhost:9090 --chrome-flags="--ignore-certificate-errors" --port=9222 --reuse-page --preset=desktop --view
```

**Comment by adamraine** [comment]
For starters, we don't officially support Edge or auditing pages in web views.

Also, the `lighthouse` command doesn't have a `--reuse-page` flag in the CLI, so you may need to set up the connection yourself using the Node API. It might be possible to connect puppeteer to the remote debugging port and then re-using the page handle with Lighthouse: https://github.com/GoogleChrome/lighthouse/blob/main/docs/puppeteer.md

*This is still uncharted territory though, I don't know how Puppeteer or Lighthouse will handle this environment even if the connection works.*
